rust/datafusion/src/physical_plan/expressions.rs

```diff
@@ -1542,20 +1574,7 @@ pub fn cast(
     if expr_type == cast_type {
         return Ok(expr.clone());
     }
-    if is_numeric(&expr_type) && (is_numeric(&cast_type) || cast_type == DataType::Utf8) {
-        Ok(Arc::new(CastExpr { expr, cast_type }))
-    } else if expr_type == DataType::Binary && cast_type == DataType::Utf8 {
-        Ok(Arc::new(CastExpr { expr, cast_type }))
-    } else if is_numeric(&expr_type)
-        && cast_type == DataType::Timestamp(TimeUnit::Nanosecond, None)
-    {
-        Ok(Arc::new(CastExpr { expr, cast_type }))
-    } else {
-        Err(ExecutionError::General(format!(
-            "Invalid CAST from {:?} to {:?}",
-            expr_type, cast_type
-        )))
-    }
+    Ok(Arc::new(CastExpr { expr, cast_type }))
```

Comment:
As in https://github.com/apache/arrow/pull/8340, I don't think partially redundant and incomplete checks for casting support in DataFusion adds much over the actual arrow casting, so I have removed the plan time checks